### PR TITLE
dvtm-unstable: init at 2018-03-31

### PR DESCRIPTION
**Pull request by luke-clifton**
###### Motivation for this change

While the project is still seeing commits coming in, there has not been a release in a while.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by luke-clifton** [comment]
1. dvtm releases are uncommon because the common use-case is to compile from source and apply patches as configuration. Since the last release, there are a few fixes for Darwin, and the patch added as part of this commit fixes a dirty buffer issue which makes it unusable with certain programs.

2. No. Would you prefer I inlined the patch?

3. The latest release doesn't work with Darwin (except with the patch I provided last time), and has dirty buffer issues in some cases. I could rejig this and submit a larger patch based of the previous release if that would be more palatable? I just thought that this was more honest...

**Comment by 7c6f434c** [comment]
I think Darwin support is usually a good enough reason to add an extra version (if there is no stable source URL for that extra patch yet, I guess inlining is a better solution), or to add a small-ish amount of small-ish patches.

As for complete replacement, I want to know @vrthra opinion as the maintainer.

It looks like by the previous upstream standards, there should have been a couple of releases since the last one, so maybe after merging PR-86 you could ask about that…

**Comment by lheckemann** [comment]
@GrahamcOfBorg build dvtm

**Comment by lheckemann** [comment]
This should be named `dvtm-<date>` rather than `dvtm-master`: https://nixos.org/nixpkgs/manual/#sec-package-naming

File: pkgs/tools/misc/dvtm/default.nix [hunk 1]
```diff
@@ -1,20 +1,29 @@
-{ stdenv, fetchurl, ncurses, customConfig ? null }:
+{ stdenv, fetchpatch, fetchurl, ncurses, customConfig ? null }:
 
 stdenv.mkDerivation rec {
 
-  name = "dvtm-0.15";
+  name = "dvtm-unstable-2018-03-31";
 
   src = fetchurl {
```
Comment (worldofpeace) [inline pkgs/tools/misc/dvtm/default.nix]: Why not use `fetchFromGitHub`?

**Comment by lheckemann** [comment]
Thanks!

**Comment by luke-clifton** [comment]
Using fetchFromGitHub now.

**Comment by lheckemann** [comment]
Just noticed another issue — since the name is changing from dvtm to dvtm-unstable, I think this update won't be caught by `nix-env -u`. It might make sense to split the unstable dvtm into its own toplevel attr.

**Comment by luke-clifton** [comment]
Ok, so, I'm just going to leave dvtm-0.15 in there if that is the consensus. I've added `dvtm-unstable` which can update faster than the release cycle. IF a new release is ever going to happen, people who use `nixpkgs.dvtm` will get it, but they won't get this update. In the meantime, people on Darwin should use `dvtm-unstable`.

**Comment by lheckemann** [comment]
The commit message [should be]( https://nixos.org/nixpkgs/manual/#submitting-changes-making-patches) "dvtm-unstable: init at 2018-03-31"

**Comment by luke-clifton** [comment]
Yes, sorry, got my repository standards mixed up.

**Pull request merged**